Source/DafnyCore/CoverageReporter/CoverageReport.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.IO;
+using System.Linq;
+using Microsoft.Dafny;
+
+namespace DafnyCore.CoverageReporter; 
+
+public class CoverageReport {
+
+  private static int nextUniqueId = 0;
+
+  // INVARIANT: CoverageSpans are sorted within each list by the position of the StartToken
+  private readonly Dictionary<string, List<CoverageSpan>> labeledFiles;
```

Comment:
I found this name confusing. Maybe `labelsByFile` or something like that would be better?

Reply:
Done!